kolibri/core/discovery/hooks.py

```diff
@@ -23,3 +23,19 @@ def on_disconnect(self, network_location):
         :type network_location: kolibri.core.discovery.models.NetworkLocation
         """
         pass
+
+
+@define_hook
+class NetworkLocationBroadcastHook(KolibriHook):
+    def on_renew(self, instance, network_locations):
```

Comment:
Do we need this anymore on the `KolibriHook`?
```
@abstractproperty
    def _not_abstract(self):
```

Seems like we're already checking if a hook is registered and whether the class is abstract, which `@register_hook` sets both registered and `_not_abstract`. I ask because the class definition does not validate that abstract properties are defined because it sees `_not_abstract` as undefined as it's patched dynamically on the class through `@register_hook`

Reply:
Hrm, possibly not, if it's not actually validating it. Could probably just default it to `False` instead and have the `@register_hook` decorator continue to set it to `True`.

Reply:
Alright I see the reason for why that was needed now. For hooks that have more than 1 method, if you don't want to require both of the methods defined, you need something abstract defined in order for the class to be viewed as abstract. So in that case, either you define those multiple methods as abstract and require every plugin hook to define them, whether or not they need an implementation or not, or you use that existing abstract property hack.

Reply:
I wish I could say that was ringing a bell with me - but it does certainly make sense.

Reply:
The main reason I guess is that in Python you cannot define a class as abstract-- a class is only abstract if it has abstract methods or properties, so this is a workaround. Slightly wonky compared to languages that have that ability to define classes abstract at the class level. I'll fix right now